test/websockex_test.exs

```diff
@@ -468,29 +468,6 @@ defmodule WebSockexTest do
     assert_receive :caught_pong
   end
 
-  test "handles a tcp message send right after connecting", context do
-    send(context.server_pid, :immediate_reply)
-
-    assert {:ok, _pid} = TestClient.start_link(context.url, %{catch_text: self()})
-
-    assert_receive {:caught_text, "Immediate Reply"}
-  end
-
-  test "handles a ssl message send right after connecting" do
-    {:ok, {server_ref, url}} = WebSockex.TestServer.start_https(self())
-
-    on_exit(fn -> WebSockex.TestServer.shutdown(server_ref) end)
-
-    {:ok, _pid} = TestClient.start_link(url, %{})
-    server_pid = WebSockex.TestServer.receive_socket_pid()
-
-    send(server_pid, :immediate_reply)
-
-    assert {:ok, _pid} = TestClient.start_link(url, %{catch_text: self()})
-
-    assert_receive {:caught_text, "Immediate Reply"}
-  end
```

Comment:
Why did we remove these two tests?

Reply:
Because the request object that is passed to the websocket handler aka. the server that implements the cowboy_websocket behavior has changed. 
From:
`{:http_req, #Port<0.1328>, :ranch_tcp, :keepalive, #PID<0.14552.0>, "GET",
 :"HTTP/1.1", {{127, 0, 0, 1}, 60570}, "localhost", :undefined, 80, "/ws",
 :undefined, "", :undefined, [],
 [
   {"host", "localhost"},
   {"connection", "Upgrade"},
   {"upgrade", "websocket"},
   {"sec-websocket-version", "13"},
   {"sec-websocket-key", "L/lP/OS6kaw+N4VTaZ8xQQ=="}
 ], [{"connection", ["upgrade"]}], :undefined, [], :waiting, "", :undefined,
 false, :waiting, [], "", :undefined}`
to:
`%{
  bindings: %{},
  body_length: 0,
  cert: :undefined,
  has_body: false,
  headers: %{
    "connection" => "Upgrade",
    "host" => "localhost",
    "sec-websocket-key" => "msu2WzAAmqsn6XroYcXc1A==",
    "sec-websocket-version" => "13",
    "upgrade" => "websocket"
  },
  host: "localhost",
  host_info: :undefined,
  method: "GET",
  path: "/ws",
  path_info: :undefined,
  peer: {{127, 0, 0, 1}, 56977},
  pid: #PID<0.334.0>,
  port: 80,
  qs: "",
  ref: #Reference<0.2532649084.3285712897.145336>,
  scheme: "http",
  sock: {{127, 0, 0, 1}, 51705},
  streamid: 1,
  version: :"HTTP/1.1"
}`

from this I think the same functionality is not possible.